cve_bin_tool/data_sources/osv_source.py

```diff
@@ -1,383 +1,269 @@
+                LOGGER.warning(f"OSV: Timeout error while fetching {url}.")
 
-            for package_data in cve_item.get("affected", []):
-                package = package_data.get("package", {})
-                if not package:
-                    continue
+    async def fetch_all(self) -> None:
+        """Concurrently fetch all zip files from OSV."""
+        LOGGER.info("OSV: Started fetching OSV CVE files...")
+        async with aiohttp.ClientSession() as session:
+            tasks = [
+                self.fetch_single(url, self.osv_path, session)
+                for url in self.download_url
+            ]
+            await asyncio.gather(*tasks)
+
+    async def extract_all(self) -> None:
+        """Extract and delete all files in the OSV cache directory."""
+        # .glob("zip") returns an iterator, so it is memory efficient to process files in the loop
+        LOGGER.info("OSV: Started extracting zip files...")
+        for file in pathlib.Path(self.osv_path).glob("*.zip"):
+            try:
+                LOGGER.debug(f"OSV: Extracting {file}")
+                with zipfile.ZipFile(file, "r") as zip_ref:
+                    zip_ref.extractall(self.osv_path)
+            except zipfile.BadZipFile:
+                LOGGER.warning(f"OSV: Error while extracting {file}.")
+            finally:
+                os.remove(file)
+                await asyncio.sleep(0.5)
+
+    @staticmethod
+    def get_formatted_data_from_json(
+        content: dict,
+    ) -> tuple[dict[str, Any], dict[str, Any] | None]:
+        cve_id, severity, vector = (
+            content.get("id"),
+            content.get("severity", None),
+            None,
+        )
 
-                product = package.get("name")
-                vendor = (
-                    "unknown"  # OSV Schema does not provide vendor names for packages
-                )
+        severity: dict | None  # type: ignore
+        if severity and "CVSS_V3" in [x["type"] for x in severity]:
+            try:
+                # Ensure the CVSS vector is valid
+                if severity[0]["score"].endswith("/"):
+                    cvss_data = CVSS3(severity[0]["score"][:-1])
+                    LOGGER.debug(f"{cve_id} : Correcting malformed CVSS3 vector.")
+                else:
+                    cvss_data = CVSS3(severity[0]["score"])
+                # Now extract CVSS attributes
+                version = "3"
+                severity = cvss_data.severities()[0]
+                score = cvss_data.scores()[0]
+                vector = cvss_data.clean_vector()
+
+            except Exception as e:
+                LOGGER.debug(e)
+                LOGGER.debug(f"{cve_id} : {severity}")
+                vector = None
+
+        cve = {
+            "ID": cve_id,
+            "severity": severity if vector is not None else "unknown",
+            "description": content.get("summary", "unknown"),
+            "score": score if vector is not None else "unknown",  # noqa
```

Comment:
(nit) Why `noqa`? An explanation would be helpful to make sure this is not something the project coding guidelines actually require. Same applies to other similar ones.